projects/Reaction/default.nix

```diff
@@ -0,0 +1,106 @@
+
+    - [x] module (in nixpkgs)
+      - [x] system.checks
+      - [x] iptables firewall.service issue
+        - [x] nixosTest assert it is successful
+      - [x] runAsRoot
+      - [x] nixosTests
+        - [x] runAsRoot
+        - [x] !runAsRoot
+        - [x] fail2ban equivalent (same as basic example test in nigpkgs?)
+    - [x] tests
+      - [x] basic example tests
+      - [x] (NO) benchmarks
+        - nginx access.log test can't be done but the other two
+        - [ ] 2 separate tests, one per benchmark?
+      - [x] nixpkgs nixosTests?
+    - [x] examples
+      - [x] basic
+      - [x] (NO) maybe googlebot and ai-crawlers
+      - But upstream is very much for maintaining examples (without tests as of now)
+    - [x] demo-vm (can't be shell)
+      - [x] tied to basic example test from above
+      - [x] basic usage/test instructions
+  */
+  nixos.modules.services = {
+    reaction = {
+      name = "Reaction";
+      module = lib.moduleLocFromOptionString "services.reaction";
+      examples.basic = {
+        module = ./examples/basic.nix;
+        description = ''
+          A setup, where reaction is run as a dedicated reaction user (indicated by runAsRoot set to false)
+
+          And permissions are explicitly given to the reaction user
+            - to be allowed to read ssh journal logs (adding reaction user to systemd-journal group)
+            - and allow changing firewall rules (adding CAP_NET_ADMIN previlige to reaciton systemd service)
+        '';
+        tests.basic.module = pkgs.nixosTests.reaction;
+      };
+    };
+  };
+
+  # benchmarks, firewall tests are not tied to an example
+  nixos.tests.benchmarks.module = import ./tests/benchmarks.nix args;
+  nixos.tests.firewall.module = pkgs.nixosTests.reaction-firewall;
+
+  nixos.demo.vm = {
+    module = ./demo/module.nix;
+    module-demo = ./demo/module-demo.nix;
+    usage-instructions = [
+      {
+        instruction = ''
+          Reaction is a very powerful rules based engine.
+          A common usecase is to scan ssh and webserver logs, and to ban hosts that cause multiple authentication errors.
+
+          An example-ssh.jsonnet file has been provided in the ngipkgs repository
+          It was copied from ''${pkgs.reaction}/share/examples/example.jsonnet and modified slightly for the purpose of showcasing the demo
+        '';
+      }
+      {
+        instruction = ''
+          - Run the demo vm, it runs an ssh server at port 10022, a user `nixos` with password `nixos` exists
+          - Run `watch journalctl -u reaction --no-pager` inside the demo vm
+
+          - Open a new terminal in your host system and run try this ssh command thrice
+          - ssh -p 10022 nixos@localhost -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
+          - Attempt login to the demo vm with wrong passwords thrice and see that you get banned
+        '';
+      }
+    ];
+    tests.demo.module = pkgs.nixosTests.reaction;
+  };
```

Comment:
Demo was failing due to `ignorecidr: ['10.0.0.0/8'],` rule specified in the example configuration, took me quite an embarassing amount of time to figure this out.

I vendored the example jsonnet configuration and left appropriate comments.

Demo works now.